main.go

```diff
@@ -152,41 +152,45 @@ scheduled backfills. No attempt is made to balance the fullness of the target
 OSDs; rather, the least busy target OSDs and PGs will be selected.
 `,
 		Args: func(cmd *cobra.Command, args []string) error {
-			if len(args) != 1 {
+			if len(args) == 0 {
 				return errors.New("a source OSD must be specified")
 			}
 
-			if _, err := strconv.Atoi(args[0]); err != nil {
-				return err
-			}
-
 			return nil
 		},
 		Run: func(cmd *cobra.Command, args []string) {
 			M = mustGetCurrentMappingState()
 
-			sourceOsd, _ := strconv.Atoi(args[0])
+			var sourceOsds []int
+			for _, s := range args {
+				osdSpecOsds := mustParseOsdSpec(s)
+				sourceOsds = append(sourceOsds, osdSpecOsds...)
+			}
 			allowMovementAcrossCrushType := mustGetString(cmd, "allow-movement-across")
 			mustParseMaxBackfillReservations(cmd)
 			mustParseMaxSourceBackfills(cmd)
-			targetOsds := mustGetOsdSpecSlice(cmd, "target-osds")
 
+			targetOsds := mustGetOsdSpecSliceMap(cmd, "target-osds")
 			tree := osdTree()
-			sourceOsdNode, ok := tree.IDToNode[sourceOsd]
-			if !ok || sourceOsdNode.Type != "osd" {
-				panic(fmt.Errorf("source OSD %d doesn't exist", sourceOsd))
-			}
 
-			for _, targetOsd := range targetOsds {
+			for targetOsd := range targetOsds {
 				targetOsdNode, ok := tree.IDToNode[targetOsd]
 				if !ok || targetOsdNode.Type != "osd" {
 					panic(fmt.Errorf("target OSD %d doesn't exist", targetOsd))
 				}
 			}
 
+			for _, osd := range sourceOsds {
+				sourceOsdNode, ok := tree.IDToNode[osd]
+				if !ok || sourceOsdNode.Type != "osd" {
+					panic(fmt.Errorf("source OSD %d doesn't exist", osd))
+				}
+				delete(targetOsds, osd)
+			}
+
```

Comment:
Make sure to document this behaviour and other changes in the command help and README.